util/topgen/validate.py

```diff
@@ -558,21 +794,28 @@ def validate_top(top, ipobjs, xbarobjs):
 
     # RV_PLIC check
 
-    # PINMUX & PADS check
-    if "padctrl" not in top:
-        log.warning("padsctrl field doesn't exist in top. Skipping pads \
-                     generation. Top input/output are directly connected from \
-                     peripherals.")
-    # Pads configuration check
-    else:
-        error += check_padctrl(top, component)
+    # Pinout, pinmux and target checks
+    # Note that these checks must happen in this order, as
+    # the pinmux and target configs depend on the pinout.
+    if "pinout" not in top:
+        log.warning("pinout field doesn't exist in top. Please consider specifying \
+                     pinout configuration.")
+        top["pinout"] = {}
+
+    error += check_pinout(top, component)
 
     if "pinmux" not in top:
         log.warning("Top {} has no 'pinmux' field. Please consider specifying \
-                        pinmux and pads configuration")
-        top["pinmux"] = OrderedDict()
-    # checking pinmux after pads as dio connects to PAD
+                        pinmux configuration.")
+        top["pinmux"] = {}
 
     error += check_pinmux(top, component)
 
+    if "targets" not in top:
+        log.warning("targets field doesn't exist in top. Please consider specifying \
+                     targets configuration.")
+        top["targets"] = {}
```

Comment:
Do we have a use case for not specifying targets? We currently always specify them, so let's make it mandatory.

Reply:
no there is currently no use case for that.
let me also make the pinout and pinmux configs mandatory - topgen would fail otherwise.